cli/src/cmd/cast/call.rs

```diff
@@ -71,7 +71,7 @@ impl CallArgs {
         let chain: Chain =
             if let Some(chain) = eth.chain { chain } else { provider.get_chainid().await?.into() };
 
-        let from = eth.sender().await;
+        let from = eth.wallet.from.unwrap_or(Address::zero());
```

Comment:
hmm,

`sender` already does this in the else branch here:

https://github.com/foundry-rs/foundry/blob/159f6299567c0712827435cf090885dcfb79c3d0/cli/src/opts/ethereum.rs#L51-L62

why do we only want to use the else branch exactly?

Reply:
ah interesting! I'm not too familiar with how the wallets/signers are implemented, but what's the difference between the if/else statement here, i.e. `self.signer.address()` vs. `self.wallet.from`? I'm guessing `self.signer(0.into()).await` is where it prompts for a keystore password, but for `cast call` and `cast estimate` we're just making a non-state changing call so we don't actually need the signer unlocked, so we shouldn't request a password

Reply:
that makes sense, but this change would now prevent using wallets entirely, which should be fine since yous can just set it as an argument if you want to.

so I think this is okay?

Reply:
As long as `self.signer.address()` and `self.wallet.from` always return the same address (guessing it’d be a bug if they didn’t) then I agree this is ok. Not using wallets seems fine  since all you need is the address, and not sure why you’d want to plug in a ledger to run `cast call` anyway 😅